internal/provider/resource/integration_aws_parameter_store.go

```diff
@@ -0,0 +1,501 @@
+	SecretPath        types.String `tfsdk:"secret_path"`
+	AWSPath           types.String `tfsdk:"parameter_store_path"`
+	AWSRegion         types.String `tfsdk:"aws_region"`
+
+	Options types.Object `tfsdk:"options"`
+}
+
+func extractOptionsParameterStore(ctx context.Context, diagnostics *diag.Diagnostics, inputOptions types.Object) (parsedOptions AwsParameterStoreMetadataStruct, hasError bool) {
+	metadata := AwsParameterStoreMetadataStruct{}
+
+	if !inputOptions.IsNull() && !inputOptions.IsUnknown() {
+		var options AwsParameterStoreOptions
+		diags := inputOptions.As(ctx, &options, basetypes.ObjectAsOptions{})
+		diagnostics.Append(diags...)
+		if diagnostics.HasError() {
+			return AwsParameterStoreMetadataStruct{}, true
+		}
+
+		// Load AWS tags into metadata
+		for _, tag := range options.AwsTags {
+			metadata.SecretAWSTag = append(metadata.SecretAWSTag, struct {
+				Key   string `json:"key,omitempty"`
+				Value string `json:"value,omitempty"`
+			}{
+				Key:   tag.Key.ValueString(),
+				Value: tag.Value.ValueString(),
+			})
+		}
+
+		metadata.ShouldDisableDelete = options.ShouldDisableDelete
+	}
+
+	return metadata, false
+}
+
+// Metadata returns the resource type name.
+func (r *IntegrationAWSParameterStoreResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_integration_aws_parameter_store"
+}
+
+// Schema defines the schema for the resource.
+func (r *IntegrationAWSParameterStoreResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Create AWS Parameter Store integration & save to Infisical. Only Machine Identity authentication is supported for this data source",
+		Attributes: map[string]schema.Attribute{
+			"options": schema.SingleNestedAttribute{
+				Description: "Integration options",
+				Optional:    true,
+				Attributes: map[string]schema.Attribute{
+					"should_disable_delete": schema.BoolAttribute{
+						Optional:    true,
+						Description: "Whether to disable deletion of existing secrets in AWS Parameter Store.",
+					},
+
+					"aws_tags": schema.ListNestedAttribute{
+						Description:   "Tags to attach to the AWS parameter store secrets.",
+						Optional:      true,
+						PlanModifiers: []planmodifier.List{listplanmodifier.RequiresReplace()},
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"key": schema.StringAttribute{
+									Description:   "The key of the tag.",
+									Optional:      true,
+									PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+								},
+								"value": schema.StringAttribute{
+									Description:   "The value of the tag.",
+									Optional:      true,
+									PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+								},
+							},
+						},
+					},
```

Comment:
let's use a SetNestedAttribute here instead so that this property is order-agnostic?